pypesto/ensemble/ensemble.py

```diff
@@ -256,6 +321,37 @@ def _compute_summary(tmp_array, percentiles_list):
         # also return the object
         return self.prediction_summary
 
+    def compute_chi2(self, amici_objective: AmiciObjective):
+        if (self.prediction_summary[MEAN] is None) or (
+                self.prediction_summary[WEIGHTED_SIGMA] is None):
+            try:
+                self.compute_summary(
+                        weighting=True,
+                        compute_weighted_sigma=True)
+            except TypeError:
+                raise ValueError('Computing a summary failed.')
+        n_conditions = len(self.prediction_results[0].conditions)
+        chi_2 = []
+        for i_cond in range(n_conditions):
+            # get measurements and put into right form
+            y_meas = amici_objective.edatas[i_cond].getObservedData()
+            y_meas = np.array(y_meas)
+            # bring into shape (n_t,n_y)
+            y_meas = y_meas.reshape(
+                (int(len(y_meas)/amici_objective.edatas[0].nytrue()),
+                 amici_objective.edatas[0].nytrue())
+            )
+            mean_traj = \
+                self.prediction_summary[MEAN].conditions[i_cond].output
+            weighted_sigmas = \
+                self.prediction_summary[
+                    WEIGHTED_SIGMA].conditions[i_cond].output
+            if y_meas.shape != mean_traj.shape:
+                raise ValueError('Shape of trajectory and shape '
+                                 'of measurements does not match.')
```

Comment:
I guess this assumes equal number of timepoints -> same timepoints. However, given the `pypesto.objective.AmiciObjective.set_custom_timepoints` method, maybe this can't be assumed. Could check timepoints explicitly.

Reply:
not sure, but shouldn't the whole optimization fail if we set custom timepoints as we could not compute a likelihood anymore? 🤔 correct me if I am wrong.

Reply:
Ah, often used for predictions rather than methods that require a likelihood.

Reply:
in that case you wouldn't compute the chi^2 value either I'd say.